softwareComponents/rofiHalSimPy/hal.cpp

```diff
@@ -0,0 +1,90 @@
+        .def_property_readonly( "maxSpeed", &Joint::maxSpeed )
+        .def_property_readonly( "minSpeed", &Joint::minSpeed )
+        .def_property_readonly( "maxTorque", &Joint::maxTorque )
+        .def_property( "velocity", &Joint::getVelocity, &Joint::setVelocity )
+        .def_property( "torque", &Joint::getTorque, &Joint::setTorque )
+        .def_property_readonly( "position", &Joint::getPosition )
+        .def( "setPosition", &Joint::setPosition )
+        .def( "onError", &Joint::onError );
+
+    py::enum_< Joint::Error >( joint, "Error" )
+        .value( "Communication", Joint::Error::Communication )
+        .value( "Hardware", Joint::Error::Hardware )
+        .export_values();
+
+    py::enum_< ConnectorPosition >( m, "ConnectorPosition" )
+        .value( "Retracted", ConnectorPosition::Retracted )
+        .value( "Extended", ConnectorPosition::Extended )
+        .export_values();
+
+    py::enum_< ConnectorOrientation >( m, "ConnectorOrientation" )
+        .value( "North", ConnectorOrientation::North )
+        .value( "East", ConnectorOrientation::East )
+        .value( "South", ConnectorOrientation::South )
+        .value( "West", ConnectorOrientation::West )
+        .export_values();
+
+    py::enum_< ConnectorLine >( m, "ConnectorLine" )
+        .value( "Internal", ConnectorLine::Internal )
+        .value( "External", ConnectorLine::External )
+        .export_values();
+
+    py::enum_< ConnectorEvent >( m, "ConnectorEvent" )
+        .value( "Connected", ConnectorEvent::Connected )
+        .value( "Disconnected", ConnectorEvent::Disconnected )
+        .value( "ConnectedPower", ConnectorEvent::ConnectedPower )
+        .value( "DisconnectedPower", ConnectorEvent::DisconnectedPower )
+        .export_values();
+
+    py::class_< ConnectorState >( m, "ConnectorState" )
+        .def_readwrite( "position", &ConnectorState::position )
+        .def_readwrite( "internal", &ConnectorState::internal )
+        .def_readwrite( "external", &ConnectorState::external )
+        .def_readwrite( "connected", &ConnectorState::connected )
+        .def_readwrite( "orientation", &ConnectorState::orientation );
+
+    py::class_< Connector >( m, "Connector" )
+        .def_property_readonly( "getState", &Connector::getState )
+        .def( "connect", &Connector::connect )
+        .def( "disconnect", &Connector::disconnect )
+        .def( "onConnectorEvent", &Connector::onConnectorEvent )
+        .def( "onPacket", &Connector::onPacket )
+        .def( "send", &Connector::send );
+        // TODO: This causes undefined symbol at runtime
+        // .def( "connectPower", &Connector::connectPower )
+        // .def( "disconnectPower", &Connector::disconnectPower );
+
+    py::class_< RoFI > rofiC( m, "RoFI" );
+    rofiC
+        .def_property_readonly( "id", &RoFI::getId )
+        .def( "getJoint", &RoFI::getJoint )
+        .def( "getConnector", &RoFI::getConnector )
+        .def( "getDescriptor", &RoFI::getDescriptor )
```

Comment:
This is a property? - I'm not sure, it just feels inconsistent with `Connector::getState`.

Maybe just have `Connector::getState` not as a property - and also keep the name (the `getState`).

But it's not much different from the `getVelocity` and other properties in the `Joint`